public/index.php

```diff
@@ -1,81 +1,87 @@
-    } else {
-        readfile(__DIR__ . '/recovery/update/maintenance.html');
+    $classLoader = require __DIR__ . '/../vendor/autoload.php';
+
+    if (!file_exists(dirname(__DIR__) . '/install.lock')) {
+        $basePath = 'recovery/install';
+        $baseURL = str_replace(basename(__FILE__), '', $context['SCRIPT_FILENAME']);
+        $baseURL = rtrim($baseURL, '/');
+        $installerURL = $baseURL . '/' . $basePath . '/index.php';
+        if (strpos($context['REQUEST_URI'], $basePath) === false) {
+            header('Location: ' . $installerURL);
+            exit;
+        }
     }
 
-    return;
-}
+    if (is_file(dirname(__DIR__) . '/files/update/update.json') || is_dir(dirname(__DIR__) . '/update-assets')) {
+        header('Content-type: text/html; charset=utf-8', true, 503);
+        header('Status: 503 Service Temporarily Unavailable');
+        header('Retry-After: 1200');
+        if (file_exists(__DIR__ . '/maintenance.html')) {
+            readfile(__DIR__ . '/maintenance.html');
+        } else {
+            readfile(__DIR__ . '/recovery/update/maintenance.html');
+        }
 
-// The check is to ensure we don't use .env if APP_ENV is defined
-if (!isset($_SERVER['APP_ENV']) && !isset($_ENV['APP_ENV'])) {
-    if (!class_exists(Dotenv::class)) {
-        throw new \RuntimeException('APP_ENV environment variable is not defined. You need to define environment variables for configuration or add "symfony/dotenv" as a Composer dependency to load variables from a .env file.');
-    }
-    $envFile = __DIR__ . '/../.env';
-    if (file_exists($envFile)) {
-        (new Dotenv())
-            ->usePutenv(true)
-            ->load($envFile);
+        exit;
     }
-}
 
-$appEnv = $_SERVER['APP_ENV'] ?? $_ENV['APP_ENV'] ?? 'dev';
-$debug = (bool) ($_SERVER['APP_DEBUG'] ?? $_ENV['APP_DEBUG'] ?? ($appEnv !== 'prod'));
+    $appEnv = $context['APP_ENV'] ?? 'dev';
+    $debug = (bool) ($context['APP_DEBUG'] ?? ($appEnv !== 'prod'));
 
-if ($debug) {
-    umask(0000);
-
-    Debug::enable();
-}
+    $trustedProxies = $context['TRUSTED_PROXIES'] ?? false;
+    if ($trustedProxies) {
+        Request::setTrustedProxies(explode(',', $trustedProxies),
+            Request::HEADER_X_FORWARDED_FOR | Request::HEADER_X_FORWARDED_PORT | Request::HEADER_X_FORWARDED_PROTO);
+    }
 
-$trustedProxies = $_SERVER['TRUSTED_PROXIES'] ?? $_ENV['TRUSTED_PROXIES'] ?? false;
-if ($trustedProxies) {
-    Request::setTrustedProxies(explode(',', $trustedProxies), Request::HEADER_X_FORWARDED_FOR | Request::HEADER_X_FORWARDED_PORT | Request::HEADER_X_FORWARDED_PROTO);
-}
+    $trustedHosts = $context['TRUSTED_HOSTS'] ?? false;
+    if ($trustedHosts) {
+        Request::setTrustedHosts(explode(',', $trustedHosts));
+    }
 
-$trustedHosts = $_SERVER['TRUSTED_HOSTS'] ?? $_ENV['TRUSTED_HOSTS'] ?? false;
-if ($trustedHosts) {
-    Request::setTrustedHosts(explode(',', $trustedHosts));
-}
+    $shopwareHttpKernel = new HttpKernel($appEnv, $debug, $classLoader);
```

Comment:
The runtime component comes with default runners. One of them is capable of handling an `HttpKernelInterface`. But since `\Shopware\Core\HttpKernel` does not implement this interface (and is in fact even incompatible with it), I have wrapped it in an anonymous class to make it compatible. This way no additional runner is needed.

Reply:
shouldn't we fix the Kernel then? :thinking:

Reply:
I would like that. But since I have no idea, why the kernels `handle` method returns an `HttpKernelResult` instead of a `Response`, I thought of this as an easier and less breaking solution. Can you tell me, why the kernel does this?

Reply:
that is a question for @OliverSkroblin :blush:

Reply:
that was required because we changed the request class in the request transformer. Without returning the "new request", the kernel terminate has not worked correctly